docs/source/advanced/test-builders.mdx

```diff
@@ -70,21 +66,18 @@ fun test() {
 }
 ```
 
-In this example we create a Human Hero with specific values for `firstName` and `age`, and the values for `lastName`
-and `height` are automatically filled with mock values.
+In this example, we create a `HumanHero` with specific values for `firstName` and `age`. The values for `lastName`and `height` are automatically filled with mock values.
```

Comment:
Couple things here:

* Is `HumanHero` correct as the name of what's created here?
* Is it obvious to Apollo Android users that it's possible to create a `HumanHero` here? Or does this rely on a codegen detail that we should maybe avoid by not using an example that includes fragments?

Reply:
Yes to the 1st question.  For the 2nd one, we could ditch the polymorphism / use of Fragment for a simpler example  -  however maybe it has value to show it in the doc as this will come up quickly in a real environment?

Reply:
One of the values of the test builders is that it's the same test code for both operation-based and response-based codegens. It's all about the GraphQL response. What about:

```
In this example, the `hero` field is a `Human` object with specific values for `firstName` and `age`. The values for `lastName`and `height` are automatically filled with mock values.
```

So overall, the advantages are:
- mocked values
- deduplicates merged fields (mostly happens with fragments)
- prefills __typename (mostly happens with fragments)

Maybe we should reorganize a bit to have the mocked values not using fragments and the rest using them?

Reply:
Okay cool I'm incorporating this rewrite suggestion for now, and we can expand this clarification in a followup PR. Making sure we hit all the content in a first pass for now